sparkly/utils.py

```diff
@@ -176,73 +176,14 @@ def parse_schema(schema):
         sparkly.exceptions.UnsupportedDataType: Cannot parse type from string: "unsupported"
     """
     field_type, args_string = re.match('(\w+)<?(.*)>?$', schema).groups()
-    args = _parse_args(args_string) if args_string else []
 
-    if field_type in ATOMIC_TYPES:
-        return ATOMIC_TYPES[field_type]()
-    elif field_type in COMPLEX_TYPES:
-        return COMPLEX_TYPES[field_type](*args)
+    if field_type in T._all_atomic_types or field_type in T._all_complex_types:
```

Comment:
why `re.match` at all, and why check the top-level `field_type`?  why not just defer entirely?

Reply:
ah you are right, so just use schema as is?